pandas/core/indexes/period.py

```diff
@@ -267,6 +269,10 @@ def __new__(cls, data=None, ordinal=None, freq=None, start=None, end=None,
         data = period.extract_ordinals(data, freq)
         return cls._from_ordinals(data, name=name, freq=freq)
 
+    @cache_readonly
```

Comment:
this might already be implemented by a superclass

Reply:
@jreback Yes. But it force convert to `int64`.
https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/index_class_helper.pxi.in#L54

Reply:
@jreback Yes, but the 1st argument is not same.
https://github.com/Licht-T/pandas/blob/6255812c30978c6a55f1758927f9c300e35cc82b/pandas/core/indexes/base.py#L1558

Reply:
I c. ok, we should make this much more generic (IOW just pass in self and have the engine extract what it needs), but that's another issue.